src/gameengine/gameobjects/abstract-game-object.ts

```diff
@@ -20,11 +22,35 @@ export abstract class AbstractGameObject {
   get id(): number {
     return this._id;
   }
+  get position(): IPosition {
+    return {...this.positions[this.positions.length - 1]};
+  }
+  set position(newPosition: IPosition) {
+    this.positions.push(newPosition);
```

Comment:
Why are both positions and requestPositions needed?

Reply:
Actually it was due to the idea of an ability to replay a run. But when I think about it know It has to be on the board level to work. I Will fix it.